server/src/main/java/org/elasticsearch/common/settings/LocallyMountedSecrets.java

```diff
@@ -191,20 +205,45 @@ LocalFileSecrets processSecretsFile(Path path) throws IOException {
         try (
             var fis = Files.newInputStream(path);
             var bis = new BufferedInputStream(fis);
-            var parser = JSON.xContent().createParser(XContentParserConfiguration.EMPTY, bis)
+            var parser = JSON.xContent()
+                .createParser(XContentParserConfiguration.EMPTY.withDeprecationHandler(DeprecationHandler.IGNORE_DEPRECATIONS), bis)
         ) {
             return secretsParser.apply(parser, null);
         }
     }
 
-    record LocalFileSecrets(Map<String, String> map, ReservedStateVersion metadata) implements Writeable {
+    record LocalFileSecrets(Map<String, byte[]> map, ReservedStateVersion metadata) implements Writeable {
+
+        public static LocalFileSecrets createLocalFileSecrets(
+            Map<String, String> stringSecrets,
+            Map<String, String> fileSecrets,
+            ReservedStateVersion metadata
+        ) {
+            Map<String, byte[]> stringSecretsMap = stringSecrets == null
+                ? Map.of()
+                : stringSecrets.entrySet()
+                    .stream()
+                    .collect(Collectors.toMap(Map.Entry::getKey, e -> e.getValue().getBytes(StandardCharsets.UTF_8)));
+
+            Map<String, byte[]> fileSecretsByteMap = fileSecrets == null
+                ? Map.of()
+                : fileSecrets.entrySet()
+                    .stream()
+                    .collect(Collectors.toMap(Map.Entry::getKey, e -> Base64.getDecoder().decode(e.getValue())));
```

Comment:
nit: It's probably cleaner to grab the decoder outside the lambda, put it into a local final, and then use it inside the lambda.